app/src/main/java/com/github/mik629/android/fundamentals/ui/movieslist/FragmentMoviesList.kt

```diff
@@ -1,72 +1,75 @@
 package com.github.mik629.android.fundamentals.ui.movieslist
 
+import android.content.Context
 import android.os.Bundle
-import android.view.LayoutInflater
 import android.view.View
-import android.view.ViewGroup
 import androidx.fragment.app.Fragment
-import com.bumptech.glide.load.resource.drawable.DrawableTransitionOptions
-import com.github.mik629.android.fundamentals.GlideApp
+import androidx.fragment.app.viewModels
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.recyclerview.widget.GridLayoutManager
+import by.kirich1409.viewbindingdelegate.viewBinding
 import com.github.mik629.android.fundamentals.R
-import com.github.mik629.android.fundamentals.data.mappers.ActorMapper
-import com.github.mik629.android.fundamentals.data.network.ServerApi
-import com.github.mik629.android.fundamentals.data.repositories.MoviesRepositoryImpl
 import com.github.mik629.android.fundamentals.databinding.FragmentMoviesListBinding
 import com.github.mik629.android.fundamentals.di.AppModule
+import com.github.mik629.android.fundamentals.di.buildGlideRequest
 import com.github.mik629.android.fundamentals.ui.global.MovieItemAdapter
 import com.github.mik629.android.fundamentals.ui.moviedetails.FragmentMovieDetails
-import com.github.mik629.android.fundamentals.vm.MoviesListViewModel
+import com.google.android.material.snackbar.BaseTransientBottomBar.LENGTH_LONG
+import com.google.android.material.snackbar.Snackbar
 
-class FragmentMoviesList : Fragment() {
-    private lateinit var binding: FragmentMoviesListBinding
+class FragmentMoviesList : Fragment(R.layout.fragment_movies_list) {
+    private val binding by viewBinding(FragmentMoviesListBinding::bind)
 
-    private val viewModel by lazy {
-        MoviesListViewModel(
-            MoviesRepositoryImpl(
-                AppModule().retrofit.create(ServerApi::class.java),
-                ActorMapper()
-            )
-        )
-    }
+    private val viewModel: MoviesListViewModel by viewModels(
+        factoryProducer = {
+            MoviesListViewModelFactory(requireContext())
```

Comment:
requireContext().applicationContext

Reply:
Done with passing appContext to AppModule